feat(sentry): use AppVersion as the default image tag

yoan-adfinis opened the pull request:
Make it easier to avoid out of sync versions in the future (with 20.7.2 using `AppVersion` should just be fine, #137)

yoan-adfinis committed:
feat(sentry): use AppVersion as the default image tag

Signed-off-by: Yoan Blanc <yoan.blanc@adfinis.com>

yoan-adfinis committed:
fixup! feat(sentry): use AppVersion as the default image tag

Signed-off-by: Yoan Blanc <yoan.blanc@adfinis.com>

DandyDeveloper commented on sentry/templates/_helper.tpl:
```diff
@@ -12,6 +12,22 @@
+{{- define "snuba.image" -}}
+{{- default "getsentry/snuba" .Values.images.snuba.repository -}}
+:
+{{- default .Chart.AppVersion .Values.images.snuba.tag -}}
```
Shouldn't this be a `tpl` with the context passed in? `tpl .Chart.AppVersion .`?

yoan-adfinis replied:
It seems to work as is.

```console
$ helm install --dry-run --generate-name --debug . | grep "getsentry\\/"
install.go:159: [debug] Original chart version: ""
install.go:176: [debug] CHART PATH: /home/yoan/soft/charts/sentry

        image: "getsentry/sentry:20.7.2"
        image: "getsentry/sentry:20.7.2"
          image: "getsentry/relay:20.7.2"
        image: "getsentry/relay:20.7.2"
        image: "getsentry/sentry:20.7.2"
        image: "getsentry/sentry:20.7.2"
        image: "getsentry/sentry:20.7.2"
        image: "getsentry/sentry:20.7.2"
        image: "getsentry/snuba:20.7.2"
        image: "getsentry/snuba:20.7.2"
        image: "getsentry/snuba:20.7.2"
        image: "getsentry/snuba:20.7.2"
```

DandyDeveloper replied:
Interesting! I also gave bad code haha, but good to know.

DandyDeveloper reviewed (commented):
I'd need to triple check, but doesn't refs like this need to be a tpl in the templating?

yoan-adfinis committed:
fixup! fixup! feat(sentry): use AppVersion as the default image tag

Signed-off-by: Yoan Blanc <yoan.blanc@adfinis.com>

hairmare committed:
fixup! fixup! fixup! feat(sentry): use AppVersion as the default image tag

Merge remote-tracking branch 'origin/develop' into image-tag-to-default-to-appversion

Signed-off-by: Lucas Bickel <lucas.bickel@adfinis.com>

Mokto commented:
Thanks! I think this is ready for merging, right?

hairmare committed:
fixup! fixup! fixup! fixup! feat(sentry): use AppVersion as the default image tag

Signed-off-by: Lucas Bickel <lucas.bickel@adfinis.com>

hairmare commented:
> Thanks! I think this is ready for merging, right?

I just added the template calls to all the new deployments and hooks from 5.1.0, so yeah I think so.

Mokto commented:
Ah, good catch! Merging now
